azurelinuxagent/common/protocol/wire.py

```diff
@@ -800,39 +795,48 @@ def update_goal_state(self, force_update=False):
             if force_update:
                 logger.info("Forcing an update of the goal state..")
 
-            if force_update or self._goal_state is None or self._goal_state.incarnation != goal_state.incarnation:
+            if force_update or \
+            self._goal_state is None or self._extensions_goal_state is None or \
+            self._goal_state.incarnation != goal_state.incarnation:
                 goal_state.fetch_full_goal_state(self)
                 self._goal_state = goal_state
                 #
                 # TODO: When Fast Track is fully enabled we should get the ExtensionsGoalState from ExtensionsConfig only
                 #       when not using fast track
                 # if not use_fast_track:
                 if self._goal_state.extensions_config_uri is None:
-                    self._extensions_goal_state = ExtensionsGoalState()
+                    self._extensions_goal_state = ExtensionsGoalState.create_empty()
                 else:
                     xml_text = self.fetch_config(self._goal_state.extensions_config_uri, self.get_header())
-                    self._extensions_goal_state = ExtensionsGoalState.from_extensions_config(xml_text)
+                    self._extensions_goal_state = ExtensionsGoalState.create_from_extensions_config(goal_state.incarnation, xml_text)
                 # END TODO
                 updated = True
 
             if use_fast_track:
-                #
-                # TODO: When Fast Track is fully enabled we should populate self._extensions_goal_state using the VmSettings
-                #
-                if self._update_vm_settings(force_update):
+                request_etag = None if force_update or self._extensions_goal_state_from_vm_settings is None else self._extensions_goal_state_from_vm_settings.get_id()
+                response_etag, vm_settings = self._fetch_vm_settings(request_etag)
+                if vm_settings is not None:
+                    # TODO: When Fast Track is fully enabled this should actually be assigned to self._extensions_goal_state
```

Comment:
We would have to verify the `created_on_ticks` before directly assigning it to `_extensions_goal_state` right? Or is that being done by the HostGA now?

Reply:
what's the verification that needs to be done?

Reply:
Ah, thanks. The HostGA does it.

In the final version the agent won't care about fabric vs fasttrack (that is handled by the hostgaplugin), the agent will only care about using the hostgaplugin (vmSettings) or the wireserver (ExtensionsConfig), the latter just for the cases where the hostgaplugin does not support vmSettings or FastTrack is disabled altogether